[needs-docs] Move diagram z-index option to a renamed Rendering tab

DelazJ opened the pull request:
Trying to standardize Layer diagram properties tab with the Labels tab:
 * Rename the Appearance tab --> Rendering tab
 * Move the Diagram z-index option from the Placement tab to Rendering; i believe it's more about visibility than x-y placement (and it's there in Labels options)

Also Improve diagram and label z-index tooltip
![image](https://user-images.githubusercontent.com/7983394/27673221-baa01ad2-5c9f-11e7-820d-68b2ed5c08af.png)


## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

@nirvn @nyalldawson ?

DelazJ committed:
[needs-docs] Move diagram z-index option to a renamed Rendering tab

Trying to standardize dialog with the Labels tab:
 * Rename the Appearance tab to Rendering
 * Move the Diagram z-index option from the placement tab to rendering; i believe it's more about visibility than x-y placement

Also Improve diagram and label z-index tooltip

nirvn commented:
@DelazJ , can you resolve the conflict in your branch? I'll try and review this early on this week.

DelazJ commented:
@nirvn Done. Thanks.

nirvn commented:
@DelazJ , are you willing to go the extra mile? 😄 I think there could be further improvements applied to harmonize the diagram UI. Check the screenshot below:

![screenshot from 2017-07-12 15-32-24](https://user-images.githubusercontent.com/1728657/28108923-68774d48-6717-11e7-83b1-400ed3e111b6.png)

I can see the following improvements needed:
- Put diagram's priority in a collapsible box to mimic label's priority UI
- Placement could be moved out of a combo box into radio boxes, like in the label's placement UI
- More generally, it'd be great for the spinboxes, text input, combo boxes, etc. to expand horizontally, just like the label UI. That way, the data-defined button nicely aligned to the right-end of the panel.

DelazJ committed:
Improve Diagram dialog

Make combobox, spinbox and textbox expand
Use group boxes in Rendering and Placement tabs

DelazJ commented:
>  are you willing to go the extra mile? :smile: I think there could be further improvements applied to harmonize the diagram UI

Here it's: the issue i can see is that we have group box with single item (not really nice according to our guidelines). Also the right alignment for widgets has been fixed (except Histogram Bar length that might need some magic).
For the placement radio buttons, i guess it'll require some code changes so will try it in another PR

![image](https://user-images.githubusercontent.com/7983394/28997895-f842a24a-7a1f-11e7-8bdf-014f8b2cdbd9.png)

I however tried to go some more yards over :), shaping the rendering tab with collapsible group boxes (like in labeling dialog - I btw found some inconsistencies in that dialog and will also propose a fix later). Let me know if it was worth the effort.

![image](https://user-images.githubusercontent.com/7983394/28997889-c291d8be-7a1f-11e7-9591-95b105b4624b.png)

nyalldawson commented on src/ui/qgsdiagrampropertiesbase.ui:
```diff
@@ -557,223 +557,400 @@
-                           </property>
-                           <property name="singleStep">
-                            <double>0.200000000000000</double>
+                            <string>Discourage labels from covering features</string>
```
Actually it's both labels AND diagrams

DelazJ replied:
Fixed, thanks

DelazJ committed:
Fix option label

nirvn commented:
@DelazJ , looking great, let me go over this one last time within the next 24 hours.

nirvn commented:
@DelazJ , all good. Thanks for the work you've done here.
